packages/mongo-nav/src/MongoNav.spec.tsx

```diff
@@ -1,81 +1,208 @@
-      await wait(() => getByTestId('project-nav'), {
-        timeout,
+
+    it('calls to fetch the data from cloud', () => {
+      expect(fetchMock.mock.calls.length).toEqual(1);
+    });
+
+    describe('when the data is returned', () => {
+      describe('and the response is valid, with fixture data', () => {
+        beforeEach(() => {
+          const fetchResponse = constructResponse({
+            jsonResponse: fixtureData, ok: true, status: 200,
+          });
+          fetchDeferred.resolve(fetchResponse as Response);
+        });
+
+        it('calls the on success handler', () => {
+          expect(onSuccessMock).toHaveBeenCalledTimes(1);
+        });
+
+        it('no longer displays the loading component', () => {
+          expect(queryByTestId(testIds.loading)).not.toBeInTheDocument();
+        });
+
+        it('displays the org nav component', () => {
+          expect(queryByTestId(testIds.orgNav)).toBeInTheDocument();
+        });
+
+        it('displays the product nav component', () => {
+          expect(queryByTestId(testIds.projectNav)).toBeInTheDocument();
+        });
       });
-      const support = getByText('Support');
-      const billing = getByText('Billing');
 
-      expect((support as HTMLAnchorElement).href).toBe(
-        `${cloudHost}/v2#/org/5d729a93/support`,
-      );
+      describe('and the response is an error code', () => {
+        beforeEach(() => {
+          const fetchResponse = constructResponse({
+            jsonResponse: fixtureData, ok: false, status: 401,
+          });
+          fetchDeferred.resolve(fetchResponse as Response);
+        });
+
+        it('calls the on error handler', () => {
+          expect(onErrorMock).toHaveBeenCalledTimes(1);
+        });
+
+        it('logs the error to the console', () => {
+          expect(consoleErrorMock).toHaveBeenCalledTimes(1);
+        });
 
-      expect((billing as HTMLAnchorElement).href).toBe(
-        `${cloudHost}/v2#/org/5d729a93/billing/overview`,
-      );
+        it('still displays the loading component', () => {
+          expect(queryByTestId(testIds.loading)).toBeInTheDocument();
+        });
+
+        it('still does not display the org or project nav components', () => {
+          expect(queryByTestId(testIds.orgNav)).not.toBeInTheDocument();
+          expect(queryByTestId(testIds.projectNav)).not.toBeInTheDocument();
+        });
+      });
     });
-  });
-  test('when mode prop is set to `dev`, fixture data is rendered inside of MongoNav', async () => {
-    await wait(() => getByTestId('project-nav'), {
-      timeout,
+
+    describe('when the data is not returned successfully', () => {
+      beforeEach(function() {
+        fetchDeferred.reject(new Error('fetch calls rarely result in a rejection'));
+      });
+
+      it('logs the error to the console', () => {
+        expect(consoleErrorMock).toHaveBeenCalledTimes(1);
+      });
+
+      it('still displays the loading component', () => {
```

Comment:
It may be premature to include this. Our loading state may not end up being an entirely different component – rather smaller differences within the existing nav. The current implementation is a placeholder until the designs are finalized.